File: front/src/Entities/Message.ts
import { Moment } from "moment"
import User from "./User"
import Room from './Room'

class Message {
    id: number
    text: string
    sender: User
    timestamp: Moment
    roomId: number

    constructor( 
        id: number,
        text: string,
        sender: User,
        timestamp: Moment,
        roomId: number
    ){
        this.id = id
        this.text = text
        this.sender = sender
        this.timestamp = timestamp
        this.roomId = roomId
    }
}

export default Message

File: Back/src/main/java/com/playful/chat/model/Message.java
package com.playful.chat.model;


import lombok.Getter;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;

@Entity
@Getter
public class Message {

    @Id
    private Long id;

    @Column(nullable = false)
    private String text;

    @Column(nullable = false)
    private LocalDateTime sendAt;

    @ManyToOne
    @JoinColumn(name = "room_id")
    private Room room;

    @ManyToOne
    @JoinColumn(name = "sender_id")
    private UserModel sender;

}